proto/api.go

```diff
@@ -79,6 +79,7 @@ type Request interface {
 	gogoproto.Message
 	// Header returns the request header.
 	Header() *RequestHeader
+	KVHeader() *KVRequestHeader
```

Comment:
There are only 27 references to the `Key`, `EndKey`, `Replica` and `RaftID` fields of `RequestHeader`. I think it would be reasonable to fix these uses to operation on a `KVRequest` instead of a `Request`. The majority are in `dist_sender.go` which isn't going to be able to send non-KV requests anyways.